build/teamcity/cockroach/nightlies/stress_impl.sh

```diff
@@ -11,6 +11,14 @@ else
     TAGS="bazel,gss,$TAGS"
 fi
 
+GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+if [ "$TC_BUILD_BRANCH" != "$GIT_BRANCH" ]; then
```

Comment:
> if [[ "$TC_BUILD_BRANCH" != "$GIT_BRANCH" ]]; then

double brackets don't work. confirmed that single brackets work as expected[1][2], i.e:

```
if [ "$TC_BUILD_BRANCH" != "$GIT_BRANCH" ]; then
   ...
```

- [1] test case 1: confirmed that TC_BUILD_BRANCH == GIT_BRANCH [works as expected](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressTriggerCeliaTestBazel/14883257?buildTab=log&focusLine=184&logView=flowAware&expandAll=true)
- [2] test case 2: confirmed that TC_BUILD_BRANCH != GIT_BRANCH [works as expected](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressTriggerCeliaTestBazel/14883256?buildTab=log&focusLine=187&logView=flowAware&expandAll=true)